crates/arrow_sql_gen/src/statement.rs

```diff
@@ -496,6 +500,9 @@ impl InsertBuilder {
         for record_batch in &self.record_batches {
             self.construct_insert_stmt(&mut insert_stmt, record_batch)?;
         }
+        if let Some(on_conflict) = on_conflict {
+            insert_stmt.on_conflict(on_conflict);
+        }
```

Comment:
After plumbing through the `sea_query::OnConflict` struct, we can apply it here. Then the next line which turns it into a string will have the correct on conflict behavior we want!